modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/managers/global/NodeCollateralStateManager.scala

```diff
@@ -0,0 +1,129 @@
+package io.constellationnetwork.node.shared.infrastructure.snapshot.managers.global
+
+import cats.effect.Async
+import cats.syntax.all._
+
+import scala.collection.immutable.{SortedMap, SortedSet}
+
+import io.constellationnetwork.node.shared.domain.nodeCollateral.UpdateNodeCollateralAcceptanceResult
+import io.constellationnetwork.schema._
+import io.constellationnetwork.schema.address.Address
+import io.constellationnetwork.schema.epoch.EpochProgress
+import io.constellationnetwork.schema.nodeCollateral._
+import io.constellationnetwork.security.Hasher
+import io.constellationnetwork.syntax.sortedCollection.sortedSetSyntax
+
+trait NodeCollateralStateManager[F[_]] {
+  def acceptNodeCollaterals(
+    lastSnapshotContext: GlobalSnapshotInfo,
+    epochProgress: EpochProgress,
+    withdrawalTimeLimit: EpochProgress
+  ): (
+    SortedMap[Address, SortedSet[NodeCollateralRecord]],
+    SortedMap[Address, SortedSet[PendingNodeCollateralWithdrawal]],
+    SortedMap[Address, SortedSet[PendingNodeCollateralWithdrawal]]
+  )
+
+  def getUpdatedCreateNodeCollaterals(
+    nodeCollateralAcceptanceResult: UpdateNodeCollateralAcceptanceResult,
+    unexpiredCreateNodeCollaterals: SortedMap[Address, SortedSet[NodeCollateralRecord]]
+  )(implicit hasher: Hasher[F]): F[SortedMap[Address, SortedSet[NodeCollateralRecord]]]
+
+  def getUpdatedWithdrawNodeCollaterals(
+    nodeCollateralAcceptanceResult: UpdateNodeCollateralAcceptanceResult,
+    unexpiredWithdrawNodeCollaterals: SortedMap[Address, SortedSet[PendingNodeCollateralWithdrawal]],
+    lastSnapshotContext: GlobalSnapshotInfo
+  )(implicit hasher: Hasher[F]): F[SortedMap[Address, SortedSet[PendingNodeCollateralWithdrawal]]]
+}
+
+object NodeCollateralStateManager {
+
+  def make[F[_]: Async](): NodeCollateralStateManager[F] = new NodeCollateralStateManager[F] {
+
+    def acceptNodeCollaterals(
+      lastSnapshotContext: GlobalSnapshotInfo,
+      epochProgress: EpochProgress,
+      withdrawalTimeLimit: EpochProgress
+    ): (
+      SortedMap[Address, SortedSet[NodeCollateralRecord]],
+      SortedMap[Address, SortedSet[PendingNodeCollateralWithdrawal]],
+      SortedMap[Address, SortedSet[PendingNodeCollateralWithdrawal]]
```

Comment:
same here as in previous comment, maybe this could be a shared class as the 
```scala
(
  existing: SortedMap[Address, A],
  unexpired: SortedMap[Address, B],
  expired: SortedMap[Address, B]
)
```
seems to be a repeated pattern